src/mame/seta/speglsht.cpp

```diff
@@ -228,68 +226,69 @@ void speglsht_state::cop_w(offs_t offset, uint32_t data, uint32_t mem_mask)
 {
 	COMBINE_DATA(&m_cop_ram[offset]);
 
-	if(m_cop_ram[offset]&0x8000) //fix (sign)
+	if (m_cop_ram[offset] & 0x8000) //fix (sign)
 	{
-		m_cop_ram[offset]|=0xffff0000;
+		m_cop_ram[offset] |= 0xffff0000;
```

Comment:
The comment is misleading, I believe it's doing a 16 to 32 sign extension?

Reply:
Yes, it's sign extension bit behavior.

Reply:
It seems pretty weird – sign-extending negative numbers but not clearing the upper bits for positive numbers.  Does the game ever write meaningful data to the upper half?  Is it just 16-bit RAM on a 32-bit bus?

Reply:
I think it's (signed) 16-bit RAM, and LSB only.